src/Symfony/Component/Asset/VersionStrategy/RemoteJsonManifestVersionStrategy.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\VersionStrategy;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Reads the versioned path of an asset from a remote JSON manifest file.
+ *
+ * For example, the manifest file might look like this:
+ *     {
+ *         "main.js": "main.abc123.js",
+ *         "css/styles.css": "css/styles.555abc.css"
+ *     }
+ *
+ * You could then ask for the version of "main.js" or "css/styles.css".
+ */
+class RemoteJsonManifestVersionStrategy implements VersionStrategyInterface
+{
+    private $manifestData;
+    private $httpResponse;
+
+    /**
+     * @param string $manifestUrl Absolute url to the manifest file
+     */
+    public function __construct(string $manifestUrl, HttpClientInterface $httpClient)
+    {
+        $this->httpResponse = $httpClient->request('GET', $manifestUrl);
+    }
+
+    /**
+     * With a manifest, we don't really know or care about what
+     * the version is. Instead, this returns the path to the
+     * versioned file.
+     */
+    public function getVersion(string $path)
+    {
+        return $this->applyVersion($path);
+    }
+
+    public function applyVersion(string $path)
+    {
+        return $this->getManifestPath($path) ?: $path;
+    }
+
+    private function getManifestPath(string $path): ?string
+    {
+        if (null === $this->manifestData) {
+            $this->manifestData = json_decode($this->httpResponse->getContent(), true);
+            if (0 < json_last_error()) {
+                throw new \RuntimeException(sprintf('Error parsing JSON from asset manifest file "%s" - %s', $this->httpResponse->getInfo()['url'], json_last_error_msg()));
+            }
+        }
+
+        return isset($this->manifestData[$path]) ? $this->manifestData[$path] : null;
```

Comment:
`??` to the rescue

Reply:
Fixed in `JsonManifestVersionStrategy` too.